### ISSUE TITLE
C varargs functions aren't supported

### ISSUE THREAD

**Issue by erg** [factor/factor#1677]
We should finish up the C FFI with this feature!

**Comment by catb0t** [comment]
Yes! I was thinking about this a while ago and I guess the only way is with Factor arrays...?

**Comment by bjourne** [comment]
Commit 86274c895e7abb27d0aefa531acb99f83165ee4f should make it work for non-Windows platforms, which is the easy part. Making it work on Windows is harder because the crazy abi requires you to duplicate float arguments and pass them in both general purpose and vector registers.
